### ISSUE TITLE
How can I instrument closed-sourced apk?

### ISSUE THREAD

**Issue by sunxiaobiu** [DroidTest/TimeMachine#15]
How can I test closed-sourced apk using TimeMachine? The instruction provides an example of open-sourced apk instrumentation, but what if the apks are closed-sourced? Considering I only have apks without source code, I am wondering if you can provide another example to indicate how the TimeMachine works with closed-sourced apk?

**Comment by sunxiaobiu** [comment]
@Allen1346

**Comment by Allen1346** [comment]
Current version of TimeMachine cannot be used to test closed-source apk, sorry about that. The earliest version of TimeMachine supported testing close-sourced apk for it used Ella( https://github.com/saswatanand/ella) as binary instrumentation tools, while unfortunately Ella didn't update for 7 years and we couldn't find a suitable substitute. We will let you know if this function is integrated someday.

**Comment by chenjshnn** [comment]
Hi @Allen1346 , do you still have the earliest version that using Ella and share with me? I am not very familiar with this domain, but want to try the tool. Thanks

**Comment by Allen1346** [comment]
Yes we do, you may check https://github.com/DroidTest/TimeMachine/tree/v1.0.0
